@here/harp-mapview/lib/MapObjectAdapter.ts

```diff
@@ -136,6 +141,10 @@ export class MapObjectAdapter {
         );
     }
 
+    get level(): number {
+        return this.object.userData.tileKey?.level;
```

Comment:
Either assert the existence of tileKey:
this.object.userData.tileKey!.level;
or the return type should be:
number|undefined

Reply:
@atomicsulfate , good spotting.

Actually, I looked a bit further, and the way I am doing this is not so great, because the `userData.tileKey` is set in the `TileGeometryCreator.ts` and this `level` getter relies on that.

I think this is for legacy reasons, so I won't touch it now, but I have added `level` as a separate parameter so I won't make it worse.